python/sglang/lang/interpreter.py

```diff
@@ -218,7 +219,10 @@ def __init__(
         self.use_thread = use_thread
         if self.use_thread:
             self.queue = queue.Queue()
-            self.worker = threading.Thread(target=self._thread_worker_func)
+
+            def _run_worker_in_context():
+                self._thread_worker_func()
+            self.worker = threading.Thread(target=contextvars.copy_context().run, args=(_run_worker_in_context, ))
```

Comment:
This is required to make tracing of the pipeline possible with Parea AI. The tracing mechanism relies on `contextvars` to create a hierarchy of function/LLM calls. By creating a new `Thread`, this context is otherwise lost.